packages/visual-editor/src/utils/applyTheme.ts

```diff
@@ -141,7 +141,7 @@ const updateFontLinksInDocument = (
   fontLinkTags: string
 ) => {
   const existingLinks = document.querySelectorAll(
-    'link[href*="fonts.googleapis.com"]'
+    'link[href*="fonts.googleapis.com"]:not([id^="visual-editor-default-fonts"]):not([id^="visual-editor-iframe-fonts"])'
```

Comment:
This only works because the passed idPrefix matches. If it's a variable and you used a different string then this would fail.